src/commands/topics.ts

```diff
@@ -166,6 +168,71 @@ async function editTopicConfig(topic: KafkaTopic): Promise<void> {
   editConfigForm.onDidDispose(() => disposable.dispose());
 }
 
+async function produceMessageFromFile(topic: KafkaTopic) {
+  if (!topic) {
+    vscode.window.showErrorMessage("No topic selected.");
+    return;
+  }
+  const options: vscode.OpenDialogOptions = {
+    canSelectMany: false,
+    openLabel: "Select JSON file",
+    filters: {
+      "JSON files": ["json"],
+    },
+  };
+
+  const fileUri = await vscode.window.showOpenDialog(options);
+  if (fileUri && fileUri[0]) {
+    const filePath = fileUri[0].fsPath;
+    const message = JSON.parse(fs.readFileSync(filePath, "utf-8"));
+
+    if (!message.key || !message.value || message.headers === 0) {
+      vscode.window.showErrorMessage(`Message must have a key, value, and headers.`);
+    }
+
+    const sidecar = await getSidecar();
+    const clusterId = topic.clusterId;
+    const connectionId = topic.connectionId;
+    const topicName = topic.name;
+
+    const recordsApi = sidecar.getRecordsV3Api(clusterId, connectionId);
+
+    const type = topic.connectionType === "LOCAL" ? undefined : "JSON";
+
+    const msgKeyAndVal = {
+      key: { data: message.key, ...(type && { type }) },
+      value: { data: message.value, ...(type && { type }) },
+    };
+
+    try {
+      await recordsApi.produceRecord({
+        topic_name: topicName,
+        cluster_id: clusterId,
+        ProduceRequest: {
+          headers: message.headers.map((header: any) => ({
+            name: header.key ? header.key : header.name,
+            value: header.value,
+          })),
+          key: msgKeyAndVal.key,
+          value: msgKeyAndVal.value,
+        },
+      });
+
+      vscode.window.showInformationMessage(`Success: Produced message to topic ${topic.name}.`);
```

Comment:
What do you think about adding a "Browse messages" button here that opens message viewer for that topic?

Reply:
I like it! https://github.com/confluentinc/vscode/issues/762